### PR TITLE
Add animation transitions and revamp code style

### PR DESCRIPTION
**Pull request by AnotherMedo**
This commit makes substantial changes to the repository and will serve as a setup for the remainder of the project development:
- Abstracted multiple composables in `app/src/main/java/com/argym/quickfix/ui/elements` to be completed/used in future composables
- Added transitions between authentications screens for smoother more pleasing and more professional viewing experience
- Added the font specified in the Figma (i.e. `Inter`) in its `ExtraBold` and `ExtraBoldItalic` versions
- Abstracted the color scheme and removed the `LoD` parameter of all composables and replacing it with its intended counterpart (c.f. `Theme.kt`)
- Abstracted text styles

This PR aims at making development less confusing and cumbersome and aid in faster and more efficient development with more reusable components. It also aims to promote ***modular development*** for future PRs 🚀 😄 

Edit: This PR ended up being more of a "save the repository" kind of thing with adding tests and fixing everything that was wrong with the repository before moving on with the project.

### CODE REVIEW

**Comment by JitterKiller** [comment]
Make sure to format your code with KTFMT, as it's currently failing the CI checks. Other than that, great work!

**Review by JitterKiller — CHANGES_REQUESTED** [review]
Make sure to format your code with KTFMT, as it's currently failing the CI checks. Other than that, great work!

**Comment by AnotherMedo** [comment]
> Make sure to format your code with KTFMT, as it's currently failing the CI checks. Other than that, great work!

These are formatting issues that were pushed in #32 they do not come from my code 😅. I must've forgot to pull before making my changes I will look into it

File: app/src/main/java/com/arygm/quickfix/ui/Announcement.kt [hunk 1]
```diff
@@ -26,56 +26,46 @@ import com.arygm.quickfix.ui.navigation.Route
 @Composable
 fun AnnouncementScreen(navigationActions: NavigationActions, isUser: Boolean, LoD: Boolean) {
 
-    val color1 = if (LoD) Color(0xFFF16138) else Color(0xFF633040)
-    val backgroundColor = if (LoD) Color.White else Color(0xFF282828)
+  val color1 = if (LoD) Color(0xFFF16138) else Color(0xFF633040)
+  val backgroundColor = if (LoD) Color.White else Color(0xFF282828)
```
Comment (daferya) [inline app/src/main/java/com/arygm/quickfix/ui/Announcement.kt:30]: The color1 and backgroundColor variables are declared twice. Move them outside the CalendarScreen function to define them globally, or place them in a separate theme configuration file if they are reused elsewhere in the app

**Review by daferya — COMMENTED** [review]
Since Scaffold is also duplicated, you can create a wrapper function that takes content and topBar as parameters. This centralizes the layout logic and prevents repetitive code
The color1 and backgroundColor variables are declared twice. Move them outside the CalendarScreen function to define them globally, or place them in a separate theme configuration file if they are reused elsewhere in the app.
To meet the coverage threshold, tests should be added for the UI components and navigation logic,

File: app/src/main/java/com/arygm/quickfix/ui/Announcement.kt [hunk 2]
```diff
@@ -26,56 +26,46 @@ import com.arygm.quickfix.ui.navigation.Route
...
+                  title = {
+                    Box(modifier = Modifier.fillMaxWidth()) {
+                      Text(
+                          text = "Announcement",
+                          color = backgroundColor,
+                          modifier = Modifier.align(Alignment.Center))
+                    }
+                  },
+                  colors = TopAppBarDefaults.topAppBarColors(containerColor = color1))
             }
```
Comment (mohagpt) [inline app/src/main/java/com/arygm/quickfix/ui/Announcement.kt:49]: The top app bar beeing present in a lot of screens can be abstracted in another file.

File: app/src/main/java/com/arygm/quickfix/ui/Calendar.kt [hunk 3]
```diff
@@ -26,56 +26,45 @@ import com.arygm.quickfix.ui.navigation.Route
...
+                  title = {
+                    Box(modifier = Modifier.fillMaxWidth()) {
+                      Text(
+                          text = "Calendar",
+                          color = backgroundColor,
+                          modifier = Modifier.align(Alignment.Center))
+                    }
+                  },
+                  colors = TopAppBarDefaults.topAppBarColors(containerColor = color1))
             }
```
Comment (mohagpt) [inline app/src/main/java/com/arygm/quickfix/ui/Calendar.kt:49]: Same here for the top app bar

**Review by mohagpt — APPROVED** [review]
> The only thing i can add is to abstract the  top app Bar in a new file, as it has been done for other components.

File: app/src/main/java/com/arygm/quickfix/ui/elements/QuickFixAnimatedBox.kt [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    rotation: Float = ANIMATED_BOX_ROTATION,
+    color: Color = colorScheme.primary,
+) {
+  Box(
+      modifier =
+          Modifier.requiredSize(size)
+              .offset(x = xOffset, y = yOffset)
+              .graphicsLayer(rotationZ = rotation)
+              .background(color)
+              .zIndex(1f))
```
Comment (Sebbayra) [inline app/src/main/java/com/arygm/quickfix/ui/elements/QuickFixAnimatedBox.kt:32]: Add a test tag for UI testing.

File: app/src/main/java/com/arygm/quickfix/ui/elements/QuickFixCheckBoxRow.kt [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+  Row(modifier = Modifier.fillMaxWidth(), verticalAlignment = Alignment.CenterVertically) {
+    Checkbox(
+        checked = checked,
+        onCheckedChange = onCheckedChange,
+        colors =
+            CheckboxDefaults.colors(
+                checkedColor = colorScheme.primary,
+                uncheckedColor = colorScheme.tertiary,
+                checkmarkColor = Color.Transparent),
+        modifier = Modifier.size(24.dp))
```
Comment (Sebbayra) [inline app/src/main/java/com/arygm/quickfix/ui/elements/QuickFixCheckBoxRow.kt:39]: Add checkbox test tag.

File: app/src/main/java/com/arygm/quickfix/ui/elements/QuickFixCheckBoxRow.kt [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    Spacer(modifier = Modifier.width(10.dp))
+    Text(label, style = MaterialTheme.typography.labelSmall, color = Color(0xFFC0C0C0))
+    Spacer(modifier = Modifier.width(4.dp))
+    Text(
+        text = underlinedText,
+        style = MaterialTheme.typography.labelSmall,
+        color = colorScheme.primary,
+        textDecoration = TextDecoration.Underline,
+        modifier = Modifier.clickable(onClick = onUnderlinedTextClick))
```
Comment (Sebbayra) [inline app/src/main/java/com/arygm/quickfix/ui/elements/QuickFixCheckBoxRow.kt:48]: Test tag here as well

File: app/src/main/java/com/arygm/quickfix/ui/elements/QuickFixCheckBoxRow.kt [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+        checked = checked,
+        onCheckedChange = onCheckedChange,
+        colors =
+            CheckboxDefaults.colors(
+                checkedColor = colorScheme.primary,
+                uncheckedColor = colorScheme.tertiary,
+                checkmarkColor = Color.Transparent),
+        modifier = Modifier.size(24.dp))
+    Spacer(modifier = Modifier.width(10.dp))
+    Text(label, style = MaterialTheme.typography.labelSmall, color = Color(0xFFC0C0C0))
```
Comment (Sebbayra) [inline app/src/main/java/com/arygm/quickfix/ui/elements/QuickFixCheckBoxRow.kt:41]: Test tag.

**Review by Sebbayra — CHANGES_REQUESTED** [review]
Overall excellent PR, the only thing i would add is the couple test tags to generate more coverage, i added them in the sub branch of testing of this branch.
We should also add the valid mail format for the login Screen.
One thing i also fixed in the sub-branch is the bug where we can press the register button in the Password screen if we just enter a valid password and keep the repeat password empty, so just approve my testing branch and this should do the job.

File: app/src/main/java/com/arygm/quickfix/ui/Announcement.kt (hunk 1 above)
Comment (AnotherMedo) [reply app/src/main/java/com/arygm/quickfix/ui/Announcement.kt]: done :)

**Comment by AnotherMedo** [comment]
> Overall excellent PR, the only thing i would add is the couple test tags to generate more coverage, i added them in the sub branch of testing of this branch. We should also add the valid mail format for the login Screen. One thing i also fixed in the sub-branch is the bug where we can press the register button in the Password screen if we just enter a valid password and keep the repeat password empty, so just approve my testing branch and this should do the job.

Solved in #40

**Review by JitterKiller — DISMISSED** [review]
Overall Great PR, this will really improve our code quality and reduce duplication code.
Great work!

**Comment by AnotherMedo** [comment]
> Overall excellent PR, the only thing i would add is the couple test tags to generate more coverage, i added them in the sub branch of testing of this branch. We should also add the valid mail format for the login Screen. One thing i also fixed in the sub-branch is the bug where we can press the register button in the Password screen if we just enter a valid password and keep the repeat password empty, so just approve my testing branch and this should do the job.

Solved in #41

**Review by Sebbayra — APPROVED** [review]
Great, everything is ready to be merged, great work.

**Review by JitterKiller — APPROVED** [review]
Great catch for the CI implementation, Sonar is now working and consistent with jaCoCo test reports.
Concerning the UI side, everything looks good to me!
This PR is now ready to be merged exeptionally (even tho we're not at 80% code coverage)

**Review by JitterKiller — APPROVED** [review]
Great catch for the CI implementation, Sonar is now working and consistent with jaCoCo test reports.
Concerning the UI side, everything looks good to me!
This PR is now ready to be merged exeptionally (even tho we're not at 80% code coverage)

**Review by Sebbayra — APPROVED** [review]
Looks, amazing, we can finally merge, thank you goat mehdi.

**Pull request merged**